substrate/frame/state-trie-migration/README.md

```diff
@@ -0,0 +1,240 @@
+
+Next, let's look at how we can do the migration using the signed submissions. For this, we have developed a basic bot in the [`polkadot-scripts`](https://github.com/paritytech/polkadot-scripts/blob/master/src/services/state_trie_migration.ts) repo.
+
+First, the signed migrations have a configuration very similar to that of automatic migration, to dictate the maximum amount of data that can be migrated in each signed transaction. This can be set using [`set_signed_max_limits`](https://github.com/paritytech/substrate/blob/77c15d2546276a865b6e8f1c5d4b1d0ec1961e72/frame/state-trie-migration/src/lib.rs#L717), which sets the storage item `SignedMigrationMaxLimits`. If not set, signed submissions are not allowed at all.
+
+This bot is just an example of how the signed migration can be done, and is a mere ~100 LoC. You are welcome to alter, fork, or re-write this bot based on your chain's need.
+
+> For example, this bot will always wait for the previous transaction to reach *full finalization*, and then submit again. You might want to change this to speed things up.
+
+The bot takes 3 arguments:
+
+1. `count`: the count of transactions to send: this is useful if you want to manually control the bot, for a short amount of time. For example, one execution will only send 10 transactions and terminate. If not set, the bot will work indefinitely until finished.
+2. `item-limit`: the limit on the number of items that the bot tries to migrate per transaction.
+3. `size-limit`: the limit on the size of the items that the bot tries to migrate per transaction.
+
+Also, the [`continue_migrate`](https://github.com/paritytech/substrate/blob/77c15d2546276a865b6e8f1c5d4b1d0ec1961e72/frame/state-trie-migration/src/lib.rs#L536) transaction, which the bot submits, expects an upper bound on the size of data migrated, aside from `item-limit`. For this, the bot will always use `item-limit * 2`.
+
+These parameters should be parsed as follows:
+
+> With each transaction, the bot is making a bet that migrating the next `item-limit` will not consume more `2 * size-limit` bytes of data.
+
+It would be futile for the bot to actually submit transactions to find the answer to this bet, since it costs money. Therefore, the bot uses the unsafe `dryRun` RPC endpoint to make sure the above assumption is met.
+
+If the `dryRun` fails, the bot will re-try with **halving the `item-limit`**. If `item-limit` reaches `0`, it means that the next key that needs to be migrated is larger than `2 * size-limit` and the bot needs to be re-executed with a larger `size-limit`. This could happen if for example the next key that needs to be migrated is e.g. the `:code:`.
+
+> Also, note that your provided `size-limit` and `item-limit` should be less than the aforementioned `SignedMigrationMaxLimits`.
+
+With all of this, let's finally see an example. `substrate-node` is configured to allow any signed account to submit the signed migration, so we don't need to alter it.
```

Comment:
Not sure if this still exists or works, but should be maybe replace with the polkadot or omni node binary.